parquet-thrift/src/main/java/parquet/thrift/struct/ThriftType.java

```diff
@@ -175,9 +180,35 @@ final public void visit(StringType stringType) {
 
     private final ThriftField[] childById;
 
+  /**
+   * Whether a struct is a union is not always known, because it was not always
+   * written to the metadata files.
+   *
+   * We should always know this in the write path, but may not in the read path.
+   */
+   public enum IsUnion {
+      YES,
+      NO,
+      UNKNOWN
+   }
```

Comment:
another possible naming for this.
enum StructType { UNION, STRUCT, UNKNOWN }
I don't feel very strongly about it.